### PR TITLE
#162633467 Implement article pagination

### PR DESCRIPTION
**Pull request by fatukunda**
**What does this PR do?**

Implements pagination functionality on articles

**Description of Task to be completed?**

- Create a pagination component.
- Create methods for paginating articles.
- Modify article actions to cater for pagination requirements

**How should this be manually tested?**

- Run `npm start` to run the application.
- Click on any item on the pagination bar to ensure that you can navigate to a different page 

**What are the relevant pivotal tracker stories?**

[#162633467](https://www.pivotaltracker.com/story/show/162633467)

### CODE REVIEW

**Comment by mubstimor** [comment]
A bit unusual to see the pagination icons at the top

File: package.json [hunk 1]
```diff
@@ -121,4 +122,4 @@
     ]
   },
   "heroku-run-build-script": true
-}
+}
```
Comment (mubstimor) [inline package.json]: newline missing here

File: .eslintrc.json [hunk 2]
```diff
@@ -50,4 +70,4 @@
       }
     ]
   }
-}
+}
```
Comment (mubstimor) [inline .eslintrc.json]: newline missing here

**Comment by mubstimor** [comment]
LGTM

**Comment by NabunyaLilian** [comment]
LGTM

File: src/reducers/articleReducer.js [hunk 3]
```diff
@@ -3,48 +3,53 @@ export const initialState = {
...
+
+  case "DELETE_ARTICLE_ERROR": {
+    return Object.assign(
+      {}, state, { errors: action.payload },
+    );
+  }
+
+  case "EDIT_ARTICLE_SUCCESS": {
+    console.log(state.articles);
+    // const updatedArticle = state.articles.map(
```
Comment (ruganda) [inline src/reducers/articleReducer.js]: Good work, but please remove the commented code

File: src/reducers/articleReducer.js [hunk 4]
```diff
@@ -3,48 +3,53 @@ export const initialState = {
...
+    //   article => ( article.id !== +action.payload.id ? article : action.payload ),
```
Comment (ruganda) [inline src/reducers/articleReducer.js]: Please remove these comments

File: src/actions/articleActions.js [hunk 5]
```diff
@@ -1,113 +1,120 @@
...
-  api.removeResource( `${ articleUrl }${ slug }/` )
-    .then( ( res ) => {
-      dispatch( deleteArticleSuccess( 'success' ) );
-    } )
-    .catch( ( err ) => {
-      console.log( err.response );
-      dispatch( deleteArticleFailure( err.response ) );
-    } );
+export const deleteArticle = slug => (dispatch) => {
+  console.log("actions");
```
Comment (ruganda) [inline src/actions/articleActions.js]: Please remove the console logs from the production code

File: src/actions/articleActions.js [hunk 6]
```diff
@@ -1,113 +1,120 @@
...
+  api.removeResource(`${articleUrl}${slug}/`)
+    .then((res) => {
+      dispatch(deleteArticleSuccess("success"));
+    })
+    .catch((err) => {
+      console.log(err.response);
```
Comment (ruganda) [inline src/actions/articleActions.js]: Remove this console log too

File: src/actions/articleActions.js [hunk 7]
```diff
@@ -1,113 +1,120 @@
...
-      dispatch( editArticleSuccess( response.data ) );
-    } )
-    .catch( ( err ) => {
-      console.log( 'error' );
-      dispatch( editArticleFailure( err.response ) );
-    } );
+export const updateArticle = (slug, articleToEdit) => (dispatch) => {
+  api.updateResource(`${articleUrl}${slug}/`, articleToEdit)
+    .then((response) => {
+      console.log("created");
```
Comment (ruganda) [inline src/actions/articleActions.js]: Remove this console log

File: src/actions/articleActions.js [hunk 8]
```diff
@@ -1,113 +1,120 @@
...
+      dispatch(editArticleSuccess(response.data));
+    })
+    .catch((err) => {
+      console.log("error");
```
Comment (ruganda) [inline src/actions/articleActions.js]: Remove the console log

File: src/reducers/articleReducer.js [hunk 9]
```diff
@@ -3,48 +3,53 @@ export const initialState = {
...
-    case 'EDIT_ARTICLE_SUCCESS': {
-      return Object.assign(
-        {}, state, { articles: state.articles },
-      );
-    }
-
-    default:
-      return state;
+  case "FETCH_ARTICLES_ERROR":
+    console.log(action.payload);
```
Comment (ruganda) [inline src/reducers/articleReducer.js]: Remove the console log

File: .eslintrc [hunk 10]
```diff
@@ -13,13 +13,28 @@
...
-    "linebreak-style": ["error", "unix"],
-    "quotes": ["error", "double"],
-    "semi": ["error", "always"],
+    "indent": [
+      "error",
+      2
+    ],
+    "linebreak-style": [
+      "error",
+      "unix"
```
Comment (ruganda) [inline .eslintrc:28]: Why do we have new rules ?

File: src/components/pagination/styles/pagination.css [hunk 11]
```diff
@@ -0,0 +1,7 @@
+.Pagination{
+    color: #38A1F3;
+    cursor: pointer;
+}
+.currentPage{
+    background-color: blue;
+}
```
Comment (ruganda) [inline src/components/pagination/styles/pagination.css]: Add a final new line

**Review by ruganda — CHANGES_REQUESTED** [review]

**Pull request merged**